Sources/Datadog/Datadog.swift

```diff
@@ -223,17 +223,28 @@ public class Datadog {
         )
 
         if let rumConfiguration = configuration.rum {
-            telemetry = RUMTelemetry(
+            let rumTelemetry = RUMTelemetry(
                 sdkVersion: configuration.common.sdkVersion,
                 applicationID: rumConfiguration.applicationID,
                 dateProvider: dateProvider,
-                dateCorrector: dateCorrector
+                dateCorrector: dateCorrector,
+                sampler: rumConfiguration.telemetrySampler
             )
 
+            telemetry = rumTelemetry
+
+            var sessionListeners: [RUMSessionListeners.Listener] = []
+            sessionListeners.append(rumTelemetry.onSessionStart)
+
+            if let sessionListener = rumConfiguration.sessionsListener {
+                sessionListeners.append(sessionListener)
+            }
+
```

Comment:
Indeed, it looks a bit clumsy. Although I like this reactive approach as it reads very well in downstream logic (_"on session change, clear telemetry"_), it appears clumsy here when it comes to configuration. It's easy to imagine that requiring a listener somewhere else (e.g. lazily) will force us back into `register()` API.

> I'm open to suggestions!

How about using the `RUMContext` which we already read in `RUMTelemetry`? It brings the current `sessionID`, so we can compare it against previous `sessionID` to determine "session change":
```swift
monitor.contextProvider.async { [weak self] context
   if context.sessionID != self.previousSessionID {
      self.previousSessionID = context.sessionID
      reset()
   }

   // ...
}
```

Downside is, it would duplicate the logic of recognising session change. Not a big deal IMO, as `sessionID` is a core element in RUM domain - it is hard to imagine that session change would ever be indicated in another way. Also, if we ever want to reuse that logic, it can be done with a higher-level API of comparing two `contexts`.

Reply:
💡 Thank you! Easier, self-contained, I like that!